mllib/src/test/scala/org/apache/spark/ml/source/image/ImageFileFormatSuite.scala

```diff
@@ -95,14 +96,14 @@ class ImageFileFormatSuite extends SparkFunSuite with MLlibTestSparkContext {
       .collect()
 
     assert(Set(result: _*) === Set(
-      Row("29.5.a_b_EGDP022204.jpg", "kittens", "2018-01"),
```

Comment:
why do we need the changes here?

Reply:
After the changes, the partition value `2018-01` became a valid date value. New partition formatter can parse it as 2018-01-01, see patterns that supported by the CAST expression:
https://github.com/apache/spark/blob/ed0e351f05ac6edc132c3a630206b2031c419e1c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala#L228
As a consequence, Spark infers DateType as the type of partition values. And finally, it converts all strings to the type.